File: db-4.8.30/csharp/DatabaseType.cs
/*-
 * See the file LICENSE for redistribution information.
 *
 * Copyright (c) 2009 Oracle.  All rights reserved.
 *
 */
using System;
using System.Collections.Generic;
using System.Text;
using BerkeleyDB.Internal;

namespace BerkeleyDB {
    /// <summary>
    /// A class representing the supported Berkeley DB access methods.
    /// </summary>
    public class DatabaseType {
        /// <summary>
        /// BTree access method
        /// </summary>
        public static readonly DatabaseType BTREE
            = new DatabaseType(DBTYPE.DB_BTREE);
        /// <summary>
        /// Hash access method
        /// </summary>
        public static readonly DatabaseType HASH
            = new DatabaseType(DBTYPE.DB_HASH);
        /// <summary>
        /// Recno access method
        /// </summary>
        public static readonly DatabaseType RECNO
            = new DatabaseType(DBTYPE.DB_RECNO);
        /// <summary>
        /// Queue access method
        /// </summary>
        public static readonly DatabaseType QUEUE
            = new DatabaseType(DBTYPE.DB_QUEUE);
        /// <summary>
        /// Unknown access method
        /// </summary>
        public static readonly DatabaseType UNKNOWN
            = new DatabaseType(DBTYPE.DB_UNKNOWN);

        private BerkeleyDB.Internal.DBTYPE dbtype;

        private DatabaseType(BerkeleyDB.Internal.DBTYPE type) {
            dbtype = type;
        }

        internal BerkeleyDB.Internal.DBTYPE getDBTYPE() {
            return dbtype;
        }

        /// <summary>
        /// Convert this instance of DatabaseType to its string representation.
        /// </summary>
        /// <returns>A string representation of this instance.</returns>
        public override string ToString() {
            switch (dbtype) {
                case DBTYPE.DB_BTREE:
                    return "BTree";
                case DBTYPE.DB_HASH:
                    return "Hash";
                case DBTYPE.DB_QUEUE:
                    return "Queue";
                case DBTYPE.DB_RECNO:
                    return "Recno";
                default:
                    return "Unknown";
            }
        }

    }
}

File: db-4.8.30/java/src/com/sleepycat/db/DatabaseType.java
/*-
 * See the file LICENSE for redistribution information.
 *
 * Copyright (c) 2002-2009 Oracle.  All rights reserved.
 *
 * $Id$
 */

package com.sleepycat.db;

import com.sleepycat.db.internal.DbConstants;

/** Database types. */
public final class DatabaseType {
    /**
    The database is a Btree.  The Btree format is a representation of a
    sorted, balanced tree structure.
    */
    public static final DatabaseType BTREE =
        new DatabaseType("BTREE", DbConstants.DB_BTREE);

    /**
    The database is a Hash.  The Hash format is an extensible, dynamic
    hashing scheme.
    */
    public static final DatabaseType HASH =
        new DatabaseType("HASH", DbConstants.DB_HASH);

    /**
    The database is a Queue.  The Queue format supports fast access to
    fixed-length records accessed sequentially or by logical record
    number.
    */
    public static final DatabaseType QUEUE =
        new DatabaseType("QUEUE", DbConstants.DB_QUEUE);

    /**
    The database is a Recno.  The Recno format supports fixed- or
    variable-length records, accessed sequentially or by logical
    record number, and optionally backed by a flat text file.
    */
    public static final DatabaseType RECNO =
        new DatabaseType("RECNO", DbConstants.DB_RECNO);

    /**
    The database type is unknown.
    */
    public static final DatabaseType UNKNOWN =
        new DatabaseType("UNKNOWN", DbConstants.DB_UNKNOWN);

    /* package */
    static DatabaseType fromInt(int type) {
        switch(type) {
        case DbConstants.DB_BTREE:
            return BTREE;
        case DbConstants.DB_HASH:
            return HASH;
        case DbConstants.DB_QUEUE:
            return QUEUE;
        case DbConstants.DB_RECNO:
            return DatabaseType.RECNO;
        case DbConstants.DB_UNKNOWN:
            return DatabaseType.UNKNOWN;
        default:
            throw new IllegalArgumentException(
                "Unknown database type: " + type);
        }
    }

    private String statusName;
    private int id;

    private DatabaseType(final String statusName, final int id) {
        this.statusName = statusName;
        this.id = id;
    }

    /* package */
    int getId() {
        return id;
    }

    /** {@inheritDoc} */
    public String toString() {
        return "DatabaseType." + statusName;
    }
}
